mantidproject/mantid
# Add Cropping Options to Eng Diff Focus Tab

~**Not ready for review until #27714 is merged**~

**Description of work.**
Cropping can now be done when focusing a run.
The GUI now allows for cropping by bank (1 or 2), or by specific spectrum numbers.

**To test:**
1. Add this line to "mantid.user.properties": `mantidqt.python_interfaces=Direct/DGS_Reduction.py Direct/DGSPlanner.py Direct/PyChop.py Direct/MSlice.py SANS/ORNL_SANS.py Utility/TofConverter.py Reflectometry/ISIS_Reflectometry_Old.py Diffraction/Engineering_Diffraction_2.py Diffraction/Powder_Diffraction_Reduction.py Utility/FilterEvents.py Diffraction/HFIR_4Circle_Reduction.py Utility/QECoverage.py SANS/ISIS_SANS.py Muon/Frequency_Domain_Analysis.py Muon/Elemental_Analysis.py Muon/Frequency_Domain_Analysis_Old.py Muon/Muon_Analysis.py` to allow access to the Engineering Diffraction 2 GUI.
1.   Open Engineering Diffraction 2 GUI
1. Switch to the Focus tab
  1. Use 305761 as the run number
  0.  Check that the combo box selector appears when the cropping check box is selected
  0.  Check selecting custom spectra reveals a Text entry
   0. Check that only sane values can be entered and error messages make sense. (1200-1400 should guarantee an output)
  0.  Check that outputs are restricted by the cropping and have sane names.

Fixes #27188

*This does not require release notes* because **the GUI is not yet visible to users.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
